py/runai_model_streamer_gcs/runai_model_streamer_gcs/credentials/credentials.py

```diff
@@ -37,7 +40,10 @@ def serialized_credentials(self) -> Dict[str, str]:
 
     def gcp_credentials(self) -> google.auth.credentials.Credentials:
         credentials = None
-        if self.credential_type == CredentialType.SERVICE_ACCOUNT_JSON:
+        endpoint_override = os.getenv(CLOUD_STORAGE_EMULATOR_ENDPOINT, default=False)
+        if endpoint_override:
+            credentials = google.auth.credentials.AnonymousCredentials()
```

Comment:
What your opinion about having an env var of `AnonymousCredentials` that user can have (maybe a parameter in the constructor), instead of using here `CLOUD_STORAGE_EMULATOR_ENDPOINT` env var.

I understand that google-cloud-cpp does use this env var so you can detect if its testing by looking at this.
But in my opinion having `AnonymousCredentials` env var will remove from the code the usage of "CLOUD_STORAGE_EMULATOR_ENDPOINT".
Personally, I believe that "production" code should not be changed for "testing" but the other way around - "testing" code should fit "production" code.

WDYT?

Reply:
Great idea. I've added an env var named `RUNAI_STREAMER_GCS_USE_ANONYMOUS_CREDENTIALS` to control this in credentials.py.